### PR TITLE
gg_images:initial commit

### PR DESCRIPTION
**Pull request by Jagroop-Singh**
PKGBUILD has been verified with systemd-nspawn container and with a vm made using blackarch-slim iso.

### CODE REVIEW

File: packages/gg_images/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,37 @@
+pkgname=gg_images
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:1]: Please follow the packaging style. Do not skip [the comments](https://github.com/BlackArch/blackarch-pkgbuilds/blob/master/PKGBUILD-go#L1)

File: packages/gg_images/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    -buildmode=pie \
+    -mod=readonly \
+    -modcacherw \
+    -ldflags "-s -w" \
+    -o $pkgname *.go
+}
+
+package(){
+	cd $pkgname
+    install -Dm 755 $pkgname "$pkgdir/usr/bin/$pkgname"
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:34]: Fix indent

File: packages/gg_images/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,37 @@
+pkgname=gg_images
+pkgver=20.fe35153
+pkgrel=1
+pkgdesc='GG profile photo downloader'
+arch=('x86_64')
+groups=('blackarch')
+url='https://codeberg.org/nanoory/GG_Images'
+license=('GPL-3.0')
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:8]:
```suggestion
license=('GPL3')
```

Use license names as found in `/usr/share/licenses/common/` wherever applicable

File: packages/gg_images/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    -modcacherw \
+    -ldflags "-s -w" \
+    -o $pkgname *.go
+}
+
+package(){
+	cd $pkgname
+    install -Dm 755 $pkgname "$pkgdir/usr/bin/$pkgname"
+	install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" README.md
+	install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:36]: You don't have to install GPL3 as it's already available in `/usr/share/licenses/common/`
 
See : https://wiki.archlinux.org/title/PKGBUILD#license and comment in [go pkgbuild template](https://github.com/BlackArch/blackarch-pkgbuilds/blob/master/PKGBUILD-go#L50)

File: packages/gg_images/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:37]: Add newline to the end.

**Review by pwnfoo — CHANGES_REQUESTED** [review]

**Comment by pwnfoo** [comment]
Hey @Jagroop-Singh, thanks a lot for the PR! 👍 

Please take some time to review our [HOWTO-DEV](https://github.com/BlackArch/blackarch/blob/master/docs/HOWTO-DEV.md)  document. This contains useful info on BlackArch development/packaging requirements.

File: packages/gg_images/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,37 @@
+pkgname=gg_images
+pkgver=20.fe35153
+pkgrel=1
+pkgdesc='GG profile photo downloader'
+arch=('x86_64')
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:5]: Any reason `aarch64` is not included?

Comment (Jagroop-Singh) [reply packages/gg_images/PKGBUILD:8]: I haven't tested it on the aarch64 architecture yet, but no reason it shouldn't work on that

File: packages/gg_images/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+pkgname=gg_images
+pkgver=20.fe35153
+pkgrel=1
+pkgdesc='GG profile photo downloader'
+arch=('x86_64' 'aarch64')
+groups=('blackarch')
+url='https://codeberg.org/nanoory/GG_Images'
+license=('GPL3')
+makedepends=('git' 'go-pie')
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:12]:
```suggestion
makedepends=('git' 'go')
```
`go-pie` is no longer required.  ` -buildmode=pie` does the trick.

**Review by pwnfoo — CHANGES_REQUESTED** [review]

File: packages/gg_images/PKGBUILD (hunk 7 above)
Comment (Jagroop-Singh) [reply packages/gg_images/PKGBUILD]: Got it, thanks for the info

File: packages/gg_images/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    -trimpath \
+    -buildmode=pie \
+    -mod=readonly \
+    -modcacherw \
+    -ldflags "-s -w" \
+    -o $pkgname *.go
+}
+
+package(){
+  cd $pkgname
```
Comment (pwnfoo) [inline packages/gg_images/PKGBUILD:35]:
```suggestion
  cd $pkgname
  
```

File: packages/gg_images/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+pkgname=gg_images
+pkgver=20.fe35153
+pkgrel=1
+pkgdesc='GG profile photo downloader'
+arch=('x86_64' 'aarch64')
+groups=('blackarch')
```
Comment (noptrix) [inline packages/gg_images/PKGBUILD:9]: @Jagroop-Singh this is the main group. this is ok and needed unless it's a big/huge prog with huge dependencies (for example: `vmcloak`, which installs virtualbox ...). can you please add a primary group after 'blackarch'? i think 'blackarch-social' and/or 'blackarch-misc' would fit here. not sure though.

File: packages/gg_images/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+url='https://codeberg.org/nanoory/GG_Images'
+license=('GPL3')
+makedepends=('git' 'go')
+source=("$pkgname::git+https://codeberg.org/nanoory/GG_Images")
+sha512sums=('SKIP')
+
+pkgver() {
+  cd $pkgname
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
```
Comment (noptrix) [inline packages/gg_images/PKGBUILD:18]: a newline here between `cd` and `echo` please.

**Review by noptrix — CHANGES_REQUESTED** [review]

**Comment by ikstream** [comment]
@Jagroop-Singh why did you close the PR?

**Comment by Jagroop-Singh** [comment]
Oh, my bad? I don't remember closing it

**Comment by Jagroop-Singh** [comment]
@ikstream Oh, I think I know where I messed up, I saw that my fork was a couple commits ahead and I reset it to blackarch master without thinking about it. I didn't realize it would close the PR, it was to start on a different pkgbuild for a different pkg.

**Comment by Jagroop-Singh** [comment]
Sorry for the mix up, I didn't think while resetting my environment.

**Comment by noptrix** [comment]
@Jagroop-Singh thanks for this!

**Review by noptrix — APPROVED** [review]

**Comment by noptrix** [comment]
@Jagroop-Singh jfyi, we do `s/_/-/g` for package directory and name whenever possible. `gg_images` will be renamed to `gg-images`. cheers

**Pull request merged**